allennlp_models/lm/modules/token_embedders/language_model.py

```diff
@@ -76,15 +77,28 @@ def __init__(
 
         # Extract the name of the tokens that the LM was trained on.
         text_field_embedder = dict_config["model"]["text_field_embedder"]
-        token_names = list(text_field_embedder["token_embedders"].keys())
-        if len(token_names) != 1:
-            # We don't currently support embedding with language models trained with multiple
-            # embedded indices.
-            #
-            # Note: We only care about embedded indices. This does not include "tokens" which
-            # is just used to compute the loss in LanguageModel.
-            raise ConfigurationError(f"LM from {archive_file} trained with multiple embedders!")
-        self._token_name = token_names[0]
+        non_empty_embedders = [
+            token_embedder_key
+            for token_embedder_key, token_embedder_value in text_field_embedder[
+                "token_embedders"
+            ].items()
+            if token_embedder_value.get("type") != "empty"
```

Comment:
This is somewhat problematic because we're basically duplicating the logic that parses the config files. There are other ways to specify an empty token embedder than this one.

Maybe we can just parse it using that same logic. I'm pretty sure you can just say `text_field_embedder = TextFieldEmbedder.from_params(Params(text_field_embedder))`, and it will create a `TextFieldEmbedder` object according to whatever the current config parsing rules are. Then you can do something like
```
non_empty_embedders = [
    name
    for name, token_embedder in text_field_embedder._token_embedders.items()
    if not isinstance(token_embedder, EmptyEmbedder)
]
```

If you want to be extra correct, you should check that `text_field_embedder` is of type `BasicTextFieldEmbedder`, though at the moment we only have that one anyways.

It's a little unfortunate that we have to read out the internal member of `BasicTextFieldEmbedder`, but it's less hacky than second-guessing the config file parser.